beacon-chain/archiver/service.go

```diff
@@ -21,31 +22,29 @@ var log = logrus.WithField("prefix", "archiver")
 // Service defining archiver functionality for persisting checkpointed
 // beacon chain information to a database backend for historical purposes.
 type Service struct {
-	ctx             context.Context
-	cancel          context.CancelFunc
-	beaconDB        db.Database
-	headFetcher     blockchain.HeadFetcher
-	newHeadNotifier blockchain.NewHeadNotifier
-	newHeadRootChan chan [32]byte
+	ctx         context.Context
+	cancel      context.CancelFunc
+	beaconDB    db.Database
+	headFetcher blockchain.HeadFetcher
+	stateFeeder blockchain.StateFeeder
 }
 
 // Config options for the archiver service.
 type Config struct {
-	BeaconDB        db.Database
-	HeadFetcher     blockchain.HeadFetcher
-	NewHeadNotifier blockchain.NewHeadNotifier
+	BeaconDB    db.Database
+	HeadFetcher blockchain.HeadFetcher
+	StateFeeder blockchain.StateFeeder
```

Comment:
Maybe StateNotifier? Something that conveys the message of notifying would be a better name

Reply:
Good name.